src/gromacs/nblib/gmxcalculator.cpp

```diff
@@ -146,4 +147,22 @@ gmx::PaddedHostVector<gmx::RVec> GmxForceCalculator::compute()
     return verletForces_;
 }
 
+void setParticlesOnGrid(std::unique_ptr<nonbonded_verlet_t>& nbv,
+                        std::vector<int>&                    particleInfoAllVdw,
+                        const std::vector<gmx::RVec>&        coordinates,
+                        const Box&                           box)
+{
+    const matrix& box_ = box.legacyMatrix;
```

Comment:
I think it is not so nice to set box_ in this way. Either this should be a temporary (renamed to box), or box.legacyMatrix should be passed directly functions, or this should be set in its own function (or the builder).

Reply:
Renamed the variable. It is a temporary one that dies after that function scope.